core/src/main/scala/org/apache/spark/sql/delta/expressions/InterleaveBits.scala

```diff
@@ -67,17 +69,49 @@ case class InterleaveBits(children: Seq[Expression])
       ints.update(i, int)
       i += 1
     }
+    InterleaveBits.interleaveBits(ints, fastInterleaveBitsEnabled)
+  }
+
+  override protected def withNewChildrenInternal(
+      newChildren: IndexedSeq[Expression]): InterleaveBits = copy(children = newChildren)
+}
+
+object InterleaveBits {
+
+  def interleaveBits(inputs: Array[Int], fastInterleaveBitsEnabled: Boolean): Array[Byte] = {
+    if (fastInterleaveBitsEnabled) {
+      inputs.length match {
+        // it's a more fast approach, use O(4 * 8)
+        // can see http://graphics.stanford.edu/~seander/bithacks.html#InterleaveTableObvious
+        case 0 => Array.empty
+        case 1 => intToByte(inputs(0))
+        case 2 => interleave2Ints(inputs(1), inputs(0))
+        case 3 => interleave3Ints(inputs(2), inputs(1), inputs(0))
+        case 4 => interleave4Ints(inputs(3), inputs(2), inputs(1), inputs(0))
+        case 5 => interleave5Ints(inputs(4), inputs(3), inputs(2), inputs(1), inputs(0))
+        case 6 => interleave6Ints(inputs(5), inputs(4), inputs(3), inputs(2), inputs(1), inputs(0))
+        case 7 => interleave7Ints(inputs(6), inputs(5), inputs(4), inputs(3), inputs(2), inputs(1),
+          inputs(0))
+        case 8 => interleave8Ints(inputs(7), inputs(6), inputs(5), inputs(4), inputs(3), inputs(2),
+          inputs(1), inputs(0))
+        case _ => defaultInterleaveBits(inputs, inputs.length)
```

Comment:
make sure there is a test case for the default case here.

Reply:
added.